app/models/user.rb

```diff
@@ -16,7 +16,7 @@ def default_values
   validates :lastname, length: { maximum: 64 }
 
   has_secure_password
-  validates :password, presence: true, length: { minimum: 6 }, format: {with: COMPLEX_PASSWORD_REGEX , message: "must be at least 6 characters and include uppercase and lowercase letters, digit and symbol."}
+  validates :password, presence: true, length: { minimum: 6 }, format: {with: COMPLEX_PASSWORD_REGEX , message: "must be at least 6 characters and include uppercase and lowercase letters, digit and symbol."}, allow_nil: true
```

Comment:
the allow_nil: true seems to be needed so you can edit your profile without having to re-enter the password